graph/appender/dead_node.go

```diff
@@ -134,3 +120,35 @@ func applyDeadNodes(trafficMap graph.TrafficMap, wkSvc business.WorkloadService,
 		s.Edges = goodEdges
 	}
 }
+
+// resolveExternalServices queries the cluster API to resolve egress services
+// by using ServiceEntries resources across all namespaces in the cluster.
+// All ServiceEntries are needed, because Istio does not distinguish where the
+// ServiceEntries are created when routing egress traffic.
+func isExternalService(service string, namespaceInfo *NamespaceInfo, globalInfo *GlobalInfo) bool {
+	if namespaceInfo.ExternalServices == nil {
+		namespaceInfo.ExternalServices = make(map[string]bool)
+
+		/* TODO: Use this call after business layer performs
+		istioCfg, err := globalInfo.Business.IstioConfig.GetIstioConfig(business.IstioConfigCriteria{
+			IncludeServiceEntries: true,
+			Namespace:             namespace,
+		})
+		checkError(err)
+		*/
+		serviceEntries, err := globalInfo.IstioClient.GetServiceEntries(namespaceInfo.Namespace)
```

Comment:
Probably I miss something here.
Better to use the business layer, instead to go to the IstioClient, no problem to add a ServiceEntries there (as busines method will enhance the model).
Also, isExternalService() sounds it iterates per service and it fetches per namespace, I wonder if it's better to fetch all services first and pass them to the isExternalService() method instead.

Reply:
You are missing the comment just above this - see line 132: "TODO: Use this call after business layer performs"

This direct access to the istio client is going to be completely removed once we confirm through testing that the business layer has the performance problems fixed in master - at that time, that commented code will be brought back (i.e use business layer).

Reply:
Ok, if this is WIP then fine, but if there is an additional fix in the business layer, let's do it there.